temporal/api/cloud/nexus/v1/message.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+package temporal.api.cloud.nexus.v1;
+
+option go_package = "go.temporal.io/api/cloud/nexus/v1;nexus";
+
+import "google/protobuf/timestamp.proto";
+
+message IncomingServiceSpec {
+    // The name of the incoming service. Must be unique within an account
+    // The name must match `[a-zA-Z_][a-zA-Z0-9_]*`.
+    string name = 1;
+
+    // The namespace to route requests to
+    string namespace = 2;
+
+    // The task queue to route requests to
+    string task_queue = 3;
+}
+
+// A binding from a service name to namespace and task queue for dispatching incoming Nexus requests.
+message IncomingService {
+    // The id of the service. This is generated by the server and is immutable
+    string id = 1;
```

Comment:
1) so there is "id" and there is "name". Do we need both for cloud? Does outgoing service reference "id" or "name" in the API (note: UX can still "hide" id concept from customer if we want)

2) Should we add an "endpoints" section? Or is namespace_name/task_queue enough to derive the appropriate information?

Reply:
We don't expose ID in the OSS API, I think we should stick with that in Cloud too for consistency.

All of the OSS endpoints use names.

Reply:
We can stick with just name, the only ask then is that the name is immutable.

The advantages of a server generated ID are:
- name can be mutable
- the ID we generate can eventually be referenced cross-account without any changes (future facing)
- Other artifacts in cloud (e.g. users) have a generated ID that is used as the primary key, so this is for consistency.
- We have severely regretted having the "user-name" tie into cloud resources in the past.

Am open to either option here, just want to discuss once beforehand

Reply:
note: if we do go the ID route, we can still try a) add GetServiceByName and b) enforce uniqueness

Reply:
TL;DR, options:

1) name only, everything goes through name. name is immutable. outgoing services reference name
2) name + id. require name uniqueness in account still. outgoing services reference id
3) name + id. no requirements on uniqueness. outgoing services reference id

OSS currently does 1. There may be benefits for 2/3 that we should discuss before we choose.

Reply:
discussed offline, will go with approach 2.